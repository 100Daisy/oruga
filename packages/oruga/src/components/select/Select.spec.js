import { shallowMount } from '@vue/test-utils'
import OSelect from '@components/select/Select'

let wrapper

describe('OSelect', () => {
    beforeEach(() => {
        wrapper = shallowMount(OSelect)
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('OSelect')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('has a select element', () => {
        expect(wrapper.contains('select')).toBeTruthy()
    })

    it('emit input event with value when value change', () => {
        wrapper.setProps({ value: true })
        expect(wrapper.vm.computedValue).toBeTruthy()
        wrapper.vm.computedValue = false
        const valueEmitted = wrapper.emitted()['input'][0]
        expect(valueEmitted).toContainEqual(false)
    })
})
