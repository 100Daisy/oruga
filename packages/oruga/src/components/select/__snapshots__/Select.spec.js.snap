// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OSelect render correctly 1`] = `
<div class="o-control-select o-control-select-icons-right"><select class="o-select">
        <!----> </select>
    <!---->
    <o-icon-stub icon="caret-down" rootclass="o-icon" clickableclass="o-icon-clickable" spinclass="o-icon-spin" both="true" class="o-icon-right"></o-icon-stub>
</div>
`;
